### PR TITLE
OTP implementation

### PR DESCRIPTION
**Pull request by bobloblaws93**
- OTP login page added
- OTP methods to access db in providers

### CODE REVIEW

File: src/pages/OTP/otp.ts [hunk 1]
```diff
@@ -0,0 +1,87 @@
...
+        this.app.getRootNav().setRoot(TabsPage);
+      }
+    });
+  }
+  OTPlogin()
+  {
+          this.app.getRootNav().setRoot(TabsPage);
+  }
+  
+  submitOTPCode(codeNumber)
```
Comment (Scieon) [inline src/pages/OTP/otp.ts]: Add tsdoc for all functions, like what you did for showAuth.

File: src/pages/OTP/otp.ts [hunk 2]
```diff
@@ -0,0 +1,87 @@
...
+      this.invalidCode = !this.invalidCode;
+      alert.present();
+     }
+      else{
+      this.app.getRootNav().setRoot(TabsPage);
+      this.invalidCode = true;
+      this.hideCodeinput = true;
+      }
+      }
+
```
Comment (Scieon) [inline src/pages/OTP/otp.ts]: Some awkward indentation in this function

**Review by Scieon — CHANGES_REQUESTED** [review]

File: .settings/.jsdtscope [hunk 3]
```diff
@@ -0,0 +1,7 @@
+<?xml version="1.0" encoding="UTF-8"?>
```
Comment (Scieon) [inline .settings/.jsdtscope:1]: What is this file, do we need it?

File: .settings/org.eclipse.wst.common.project.facet.core.xml [hunk 4]
```diff
@@ -0,0 +1,4 @@
+<?xml version="1.0" encoding="UTF-8"?>
```
Comment (Scieon) [inline .settings/org.eclipse.wst.common.project.facet.core.xml:1]: Same comment as above

File: .settings/org.eclipse.wst.jsdt.ui.superType.container [hunk 5]
```diff
@@ -0,0 +1 @@
+org.eclipse.wst.jsdt.launching.JRE_CONTAINER
```
Comment (Scieon) [inline .settings/org.eclipse.wst.jsdt.ui.superType.container:1]: ?

**Review by Scieon — CHANGES_REQUESTED** [review]

**Comment by ghiobi** [comment]
Sorry @ronnpang1 there are a few things to consider here.
1. Before you always commit please make sure you don't add any necessary files.
2. I think you got the general idea of OTP pretty wrong.
3. First research what is OTP before implementing and ask questions with your teammates.
4. Please don't copy code in this project, because that usually doesn't work.

**Comment by ghiobi** [comment]
@ronnpang1 will you update your changes? If you need help on that, let us know immediately.

**Comment by ghiobi** [comment]
@ronnpang1 Is this complete? please fix the issues/conflicts you were told to do. If you need help please leave some comments.

**Comment by ghiobi** [comment]
Pull request https://github.com/Project-Epsilon/mobile/pull/26 will replace this. thank you for your time.

> Success is the result of perfection, hard work, learning from failure, loyalty, and persistence. - Colin Powell

**Pull request closed without merging**